infra/Pos/Network/Types.hs

```diff
@@ -211,3 +218,52 @@ initQueue NetworkConfig{..} = do
     enqueuePolicy = OQ.defaultEnqueuePolicy ourNodeType
     dequeuePolicy = OQ.defaultDequeuePolicy ourNodeType
     failurePolicy = OQ.defaultFailurePolicy ourNodeType
+
+{-------------------------------------------------------------------------------
+  Auxiliary
+-------------------------------------------------------------------------------}
+
+type Resolver = DNS.Domain -> IO (Either DNSError [IPv4])
+
+-- | Initialize the DNS library only when we need to
```

Comment:
Or presumably we could have done it by checking if every host is specified as a numeric address and just split the cases in two and dealing with the general case where we need DNS as now, and have a special (simple) case for all numeric hosts (which should be simple since it's pure and cannot fail). That might be less code than doing this trickery.

Or just say that we don't care about the performance of the DNS stuff an re-init it each time. Where we're using it, it is not performance sensitive. I'm not sure it's worth this complexity with concurrency tricks.